metadata_service/proxy/atlas_proxy.py

```diff
@@ -168,57 +171,93 @@ def _get_user_entity(self, user_id: str) -> EntityUniqueAttribute:
             raise NotFoundException('(User {user_id}) does not exist'
                                     .format(user_id=user_id))
 
-    def _create_reader(self, table_entity: EntityUniqueAttribute, user_guid: str, reader_qn: str) -> None:
+    def _create_bookmark(self, entity: EntityUniqueAttribute, user_guid: str, bookmark_qn: str,
+                         table_uri: str) -> None:
         """
-        Creates a reader entity for a specific user and table uri.
+        Creates a bookmark entity for a specific user and table uri.
         :param user_guid: User's guid
-        :param reader_qn: Reader qualifiedName
+        :param bookmark_qn: Bookmark qualifiedName
         :return:
         """
-        reader_entity = {
-            'typeName': self.READER_TYPE,
-            'attributes': {'qualifiedName': reader_qn,
-                           'isFollowing': True,
-                           'count': 0,
-                           'user': {'guid': user_guid}}
+
+        bookmark_entity = {
+            'entity': {
+                'typeName': self.BOOKMARK_TYPE,
+                'attributes': {'qualifiedName': bookmark_qn,
+                               self.BOOKMARK_ACTIVE_KEY: True,
+                               'entityUri': table_uri,
+                               'user': {'guid': user_guid},
+                               'entity': {'guid': entity.entity[self.GUID_KEY]}}
+            }
         }
 
-        self._driver.entity_post.create(data=reader_entity)
-        table_entity.entity[self.ATTRS_KEY]['readers'].append(reader_entity)
-        table_entity.update()
+        self._driver.entity_post.create(data=bookmark_entity)
 
-    def _get_reader_entity(self, table_uri: str, user_id: str) -> EntityUniqueAttribute:
+    def user_exists(self, user_id: str) -> bool:
+        """
+        Verifies if given user exists in Atlas.
+        :param user_id: User's id
+        :return: True if user exists, otherwise False
         """
-        Fetch a Reader entity from parsing table uri and user id.
-        If Reader is not present, create one for the user.
+        user_entity = self._driver.entity_unique_attribute(self.USER_TYPE, qualifiedName=user_id)
+
+        if not user_entity.entity:
+            return False
+        else:
+            return True
+
+    def create_user(self, user_id: str) -> None:
```

Comment:
what does this function do ? where it is being used ?

Reply:
I added it as a helper method and am using in in our `get_user_details` function in metadata config